### PR TITLE
Fix new project root path not being used in CLI-launched GUI

### LINKED ISSUE #408: [BUG] `exo gui` outside of a workspace no longer respects default root path
Go to a non-workspace directory and run `exo gui`, it will take you to the new project page and provide a `root=` query parameter. The GUI no longer makes use of this parameter as a default value when creating a project.

It was even more broken in that it just 404'd, but that is fixed here: https://github.com/deref/exo/commit/d92d1e40e8d5367a4e1a6587b6eb0c66110b24ac

### PR DESCRIPTION
**Pull request by jwmza**
Fixes #408

### CODE REVIEW

**Comment by jwmza** [comment]
Tested & working both when there is and isn't a root parameter given.

Notable: currently this will prompt you to create the New Project as a folder within the folder from which you launched, rather than creating a new project with the folder itself. This might not be ideal but it's a separate issue - if you give no folder name, it should use the working directory as the root of the new project.

File: gui/src/pages/NewProject.svelte [hunk 1]
```diff
@@ -45,7 +48,11 @@
                 .slice(0, search.length) === search.toLocaleLowerCase()) as template}
           <button
             on:click={() => {
-              router.push(`#/new-project/${template.name}`);
+              router.push(
+                `#/new-project/${template.name}${
```
Comment (brandonbloom) [inline gui/src/pages/NewProject.svelte:52]: Even though we control the template name, we should code defensively here and use an encodeURIComponent on it.

File: gui/src/pages/NewProjectConfig.svelte [hunk 2]
```diff
@@ -28,6 +31,9 @@
   let dirPromise: Promise<ReadDirResult> = api.kernel
     .getUserHomeDir()
     .then((dir) => {
+      if (root) {
+        dir = root;
+      }
       setWorkingDirectory(dir);
       homeDirectory = dir;
       return dirPromise;
```
Comment (brandonbloom) [inline gui/src/pages/NewProjectConfig.svelte:39]: This is really strange... In general, the side effects in this callback seem like they are subverting the Svelte reactive mechanisms.

**Review by brandonbloom — APPROVED** [review]

File: gui/src/pages/NewProjectConfig.svelte (hunk 2 above)
Comment (jwmza) [reply gui/src/pages/NewProjectConfig.svelte:39]: Do you man the new 3 lines or the entire mechanism @brandonbloom

Comment (brandonbloom) [reply gui/src/pages/NewProjectConfig.svelte:39]: The side effects in this callback in general.

**Pull request merged**